src/node/transaction.cpp

```diff
@@ -32,7 +32,7 @@ static TransactionError HandleATMPError(const TxValidationState& state, std::str
 
 TransactionError BroadcastTransaction(NodeContext& node, const CTransactionRef tx, std::string& err_string, const CAmount& max_tx_fee, bool relay, bool wait_callback)
 {
-    // BroadcastTransaction can be called by either sendrawtransaction RPC or the wallet.
+    // BroadcastTransaction can be called by sendrawtransaction RPC, submitpackage RPC or the wallet.
```

Comment:
I don't think we care about which specific RPCs call it? Would just mention that it can be called by RPC, that's the key info - and more maintainable.

```suggestion
    // BroadcastTransaction can be called by RPC or by the wallet.
```

Reply:
Extending that logic, should we document all the call sites of each helper function that's used by RPCs because it can be helpful sometimes? That seems abnormal to me. I understood the only relevance of this line to be in relation to the next one: "chainman, mempool and peerman are initialized before the RPC server and wallet are started". For this, it doesn't matter which RPCs call it.

Reply:
`BroadcastTransaction()` is not just any helper function, it may irreversibly affect user's privacy.

Reply:
I think `BroadcastTransaction can be called by RPC or by the wallet` is ambiguous in this case. 
By which RPC?

`BroadcastTransaction` can be invoked by other components that will like to submit transactions to the mempool and relay them to peers, in addition to the wallet and RPC. 
It's accessible through the node interface https://github.com/bitcoin/bitcoin/blob/717103bccec8d271f61f9cd6481b334bd9889146/src/node/interfaces.cpp#L341

If I understand the rationale behind this comment correctly, it's specifically indicating who is currently calling the function, that is `sendrawtransaction`, `submitpackage` RPC, and the wallet

But if the rationale is indicating the generic meaning of who can invoke `BroadcastTransaction`, then yes, I agree it would be even more maintainable to simply delete the comment?

Reply:
Deleted the comment, and updated the PR description.

Reply:
You've deleted the comment entirely?

Reply:
I don't understand why we'd delete it entirely either. It belongs together with the next sentence. I suppose it can be deduced from it too, but the current approach doesn't seem like an improvement to me. I still think my suggested phrasing works best from what I've seen so far. I don't think it's ambiguous, it just doesn't specify which RPC is called, which is irrelevant to the rest of the documentation and as such we shouldn't make it look like it is.

Reply:
Yes I agree with your point.
Reverting back with your initial suggestion.

Thanks @stickies-v